decompiler_agent/src/main/java/org/jrd/agent/api/Variables.java

```diff
@@ -0,0 +1,95 @@
+package org.jrd.agent.api;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Variables {
+
+    /**
+     * we have to have all inernal classes inited, so any later classes can use them without manual INIT.
+     */
+    public static void init() {
+        NoSuchFakeFieldException.init();
+        Global.init();
+    }
+
+    protected Variables() {
+        // prevents calls from subclass... checkstile sucks
+        throw new UnsupportedOperationException();
+    }
+
+    public static class NoSuchFakeFieldException extends RuntimeException {
+
+        public NoSuchFakeFieldException() {
+        }
+
+        public NoSuchFakeFieldException(String s) {
+            super(s);
+        }
+
+        public static void init() {
+        }
+    }
+
+    public static class Global {
+
+        protected Global() {
+            // prevents calls from subclass... checkstile sucks
+            throw new UnsupportedOperationException();
+        }
+
+        private static final Map<String, Object> GLOBALS = Collections.synchronizedMap(new HashMap<>());
+
+        public static void init() {
+        }
+
+        public static Object set(String name, Object value) {
+            Object old = GLOBALS.put(name, value);
+            if ("jrd.debug".equals(old)) {
+                System.err.println("We do nto care about old");
+            }
+            return value;
+        }
+
+        public static Object setNoReplace(String name, Object value) throws NoSuchFakeFieldException {
+            if (GLOBALS.containsKey(name)) {
+                throw new NoSuchFakeFieldException();
+            } else {
+                Object old = GLOBALS.put(name, value);
+                if (old != null) {
+                    throw new NoSuchFakeFieldException("Map is broken!");
+                }
+                return value;
+            }
+        }
+
+        public static Object get(String name) throws NoSuchFakeFieldException {
+            if (GLOBALS.containsKey(name)) {
+                return GLOBALS.get(name);
+            } else {
+                throw new NoSuchFakeFieldException();
+            }
+        }
+
+        public static Object getOrCrete(String name, Object defaulValue) {
+            if (GLOBALS.containsKey(name)) {
+                return GLOBALS.get(name);
+            } else {
+                Object old = GLOBALS.put(name, defaulValue);
+                if (old != null) {
+                    throw new NoSuchFakeFieldException("Map is broken!");
```

Comment:
Repeat from line#61.

Reply:
same reason. Highlgting we are returning new, instead old